api/v1/qdrantcluster_types.go

```diff
@@ -15,8 +15,22 @@ const (
 )
 
 const (
-	// The Annotation key to trigger a restart.
+	// RestartedAtAnnotationKey is the annotation key to trigger a restart.
+	// The annotation should be placed on the QdrantCluster instance.
+	// The value should be a [RFC3339 formatted] date.
+	// If the value is updated it will retrigger the restart.
+	// The operator will not remove the annotation once restarted.
+	// For historical reasons the key doesn't start with `operator.qdrant.com/`
 	RestartedAtAnnotationKey = "restartedAt"
+	// RecreateNodeAnnotationKey is the annotation key to recreate a certain node.
+	// The annotation should be placed on the QdrantCluster instance.
+	// Multiple annotations can be added, the operator will handle them all.
+	// The value should contain the node index to be replaced (e.g. "3").
+	// The node index should be one of the values specified in .Status.AvailableNodeIndexes.
+	// If a none existing node index is provided, the operator will remove the annotation (with a warning in the log).
+	// Once the operator has replaced the node it will remove the annotation.
```

Comment:
Operator should not remove the annotation, instead it should reflect the outcome of the requested operation at `.status`

Given that we would like to restart a node, adding this annotation to a pod (as original suggested by @bashofmann ) is a more natural option.

> what if the pod is crashing (and be deleted by k8s) and constantly re-created (by the operator). It will be hard to to add an annotation is this case, can't we place an annotation on the QdrantCluster (which can be removed by the operator if it handled the operation)?

If pod is crashing, containers inside the pod will be restarted, but pod resource itself is not going to removed:

https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/

If it crashed constantly, k8s stops eventually to try to restart it and mark its status as `CrashLoopBackoff`.

Thus it is totally doable to add annotations/update spec of a pod so that it can be restarted again by k8s.

Reply:
Added the annot to pod instead of the QC.